lib/publiccloud/eks.pm

```diff
@@ -0,0 +1,50 @@
+# SUSE's openQA tests
+#
+# Copyright © 2021 SUSE LLC
+#
+# Copying and distribution of this file, with or without modification,
+# are permitted in any medium without royalty provided the copyright
+# notice and this notice are preserved.  This file is offered as-is,
+# without any warranty.
+
+# Summary: Helper class for Amazon Elastic Container Registry (ECR)
+#
+# Maintainer: Ivan Lausuch <ilausuch@suse.de>, qa-c team <qa-c@suse.de>
+# Documentation: https://docs.aws.amazon.com/AmazonECR/latest/userguide/docker-push-ecr-image.html
+
+package publiccloud::eks;
+use Mojo::Base 'publiccloud::aws';
+use testapi;
+
+=head2 push_container_imgage
+Upload a container image to the ECR. Required parameter is the
+name of the image, previously stored in the local registry. And
+the tag (name) in the public cloud containers repository
+Retrieves the full name of the uploaded image or die.
+=cut
+sub push_container_imgage {
+    my ($self, $image, $tag) = @_;
+
+    my $region     //= $self->region();
```

Comment:
/sarcasm mode on: perltidy is so funny... it doesn't complain that this is not in perfect alignment with lines below :D